runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3178,15 +3178,50 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 {
 	MM_GCExtensions* extensions = MM_GCExtensions::getExtensions(vmThread);
 	J9JavaVM *vm = vmThread->javaVM;
-	bool result = true;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
-
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+	/* Note here we update this parameter which represents the machine physical memory,
+	 * but not the original heap geometry from the snapshot run. We just force the heap
+	 * not to go beyond a certain thereshold calculated based on the new (restore) memory
+	 * availalibity through setting SoftMx.
+	 */
+	extensions->usablePhysicalMemory = omrsysinfo_get_addressable_physical_memory();
+	if (!extensions->userSpecifiedParameters._Xmx._wasSpecified){
+		uintptr_t candidateSoftMx = 0;
+		if (0.0 >= extensions->maxRAMPercent) {
```

Comment:
The condition should be ">", not ">=" as far as 0.0 is allowed value to be provided in the option. It will fail obviously, but it is expected behaviour.
Also would you mind to invert logical condition (as `(0.0 <= extensions->maxRAMPercent`)) and swap halves of this if statement for easier reading (I know it might be my personal preference, I think order "if (option_provided) { <...>} else {<...>} is more logical.

Reply:
Will it be illegal for a user to set -XX:MaxRAMPercentage=0.0?